buildSrc/src/main/kotlin/JarCompressing.kt

```diff
@@ -0,0 +1,102 @@
+import groovy.json.JsonOutput
+import groovy.json.JsonSlurper
+import org.gradle.kotlin.dsl.support.uppercaseFirstChar
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.tree.ClassNode
+import java.io.File
+import java.util.jar.JarEntry
+import java.util.jar.JarFile
+import java.util.jar.JarOutputStream
+import java.util.zip.Deflater
+
+enum class CompressionType(val id: Int?) {
```

Comment:
Not sure why `id` is needed

Reply:
`id` is what advzip uses as its compression flags. although I could just use ordinals, this is probably safer.

Reply:
Oh, that makes sense then, nevermind.